src/ImageSharp/Formats/Jpeg/Components/Decoder/JpegBitReader.cs

```diff
@@ -212,7 +212,7 @@ public bool FindNextMarker()
     private int ReadStream()
     {
         int value = this.badData ? 0 : this.stream.ReadByte();
-        if (value == -1)
+        if (value == -1 || (this.badData && this.data == 0 && this.stream.Position >= this.stream.Length))
```

Comment:
Can describe the stuff behind `||` in a comment?

Reply:
Done